test/map/test_core.py

```diff
@@ -263,6 +263,25 @@ def square(x: DataPoint) -> DataPoint:
         self.assertIsNone(x21_mapped)
         self.assertEqual(square_hit_tracker.n_hits, 1)
 
+    def test_decorator_mapper_memoized_use_memoize_key(self) -> None:
+        square_hit_tracker = SquareHitTracker()
+
+        @lambda_mapper(memoize=True, memoize_key=lambda x: x.num)
+        def square(x: DataPoint) -> DataPoint:
+            x.num_squared = square_hit_tracker(x.num)
+            return x
+
+        x8 = self._get_x()
```

Comment:
It might be preferable to have an example of canonical usage in this unit test. For example, using a custom unique ID:

```
x1 = SimpleNamespace(uid="id1", num=8, unhashable=some_unhashable_pandas_object)
x2 = SimpleNamespace(uid="id1", num=8, unhashable=some_unhashable_pandas_object)
...
```

and then we use `memoize_key = lambda x: x.uid`

Reply:
Great idea.

Reply:
@Wirg let me know when you can make this change, then happy to approve the PR! Thanks again for your patience!

Reply:
So I did the change. I am not fully satisfied. If tomorrow someone change :
- `get_hashable` to support pandas dataframe
- `memoize_key` not to be used

The test won't fail.